tools/swift-format/Sources/Rules/ValidateDocumentationComments.swift

```diff
@@ -11,5 +11,112 @@ import SwiftSyntax
 
+    // If a single sentence summary is the only documentation, parameter(s) and
+    // returns tags may be ommitted.
+    if commentInfo.oneSentenceSummary != nil &&
+      commentInfo.commentParagraphs!.isEmpty &&
+      params.isEmpty &&
+      commentInfo.returnsDesc == nil {
+      return
+    }
+
+    // Indicates if the documentation uses 'Parameters' as description of the
+    // documented parameters.
+    let hasPluralDesc = declComment.components(separatedBy: .newlines)
+      .contains { $0.trimmingCharacters(in: .whitespaces).starts(with: "- Parameters") }
+
+    validateReturn(node, returnDesc: commentInfo.returnsDesc)
+    let funcParameters = funcParametersIdentifiers(in: node.signature.input.parameterList)
+
+    // If the documentation of the parameters is wrong 'docCommentInfo' won't
+    // parse the parameters correctly. First the documentation has to be fix
+    // in order to validate the other conditions.
+    if hasPluralDesc && funcParameters.count == 1 {
+      diagnose(.useSingularParameter, on: node)
+      return
+    }
+    else if !hasPluralDesc && funcParameters.count > 1 {
+      diagnose(.usePluralParameters, on: node)
+      return
+    }
+
+    // Ensures that the parameters of the documantation and the function signature
+    // are the same.
+    if (params.count != funcParameters.count) ||
+      !parametersAreEqual(params: params, funcParam: funcParameters) {
+      diagnose(.parametersDontMatch(funcName: node.identifier.text), on: node)
+    }
+  }
+
+  /// Ensures the function has a return documentation if it actually returns
+  /// a value.
+  func validateReturn(_ node: FunctionDeclSyntax, returnDesc: String?) {
+    if node.signature.output == nil && returnDesc != nil {
+      diagnose(.removeReturnComment(funcName: node.identifier.text), on: node)
+    }
+    else if node.signature.output != nil && returnDesc == nil {
+      diagnose(.documentReturnValue(funcName: node.identifier.text), on: node)
+    }
+  }
+}
+
+/// Iterates through every parameter of paramList and returns a list of the
+/// paramters identifiers.
+func funcParametersIdentifiers(in paramList: FunctionParameterListSyntax) -> [String] {
+  var funcParameters = [String]()
+  for parameter in paramList
+  {
+    guard let parameterIdentifier = parameter.firstName else { continue }
+    funcParameters.append(parameterIdentifier.text)
+  }
+  return funcParameters
+}
+
+/// Indicates if the parameters name from the documentation and the parameters
+/// from the declaration are the same.
+func parametersAreEqual(params: [parameter], funcParam: [String]) -> Bool {
+  for index in 0..<params.count {
+    if params[index].paramName != funcParam[index] {
+      return false
+    }
+  }
+  return true
+}
+
+extension Diagnostic.Message {
+  static func documentReturnValue(funcName: String) -> Diagnostic.Message {
+    return Diagnostic.Message(.warning, "Document the return value of \(funcName) .")
```

Comment:
IIRC, diagnostics should start with lowercase letters (not like an actual sentence) and not have a period at the end; please make those changes throughout.